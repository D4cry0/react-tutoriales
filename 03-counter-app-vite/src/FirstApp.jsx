import PropTypes from 'prop-types';

import { Fragment } from "react"

// Fragment se puede importar 
// <Fragment></Fragment> = <></>
// Es una forma rapida de llamar fragmentos <></>

const arr = [1,2,3,4,5,6,7,8,9];

const nObj = {
    msj: 'Holaaa',
    nom: 'Arny'
}

const getSaludo = (nombre) => {
    return `Hola ${nombre} es un buen dÃ­a`;
}

// Los elementos no se pueden "imprimir" eso aplica
// para las funciones asyncronas

// export const FirstApp = ( props ) => {
// propierties - son para comunicarse entre componentes
// se pueden desestructurar

// propTypes
// para forzar el tipo de datos y que se envien

export const FirstApp = ({ title, numero, nombre }) => {
    return (
        <>
            <h1 data-testid="test-title">{ title }</h1>
            <p>Soy un nuevo texto</p>
            <p>Soy una variable { 1+1 } </p>
            <p>{ arr }</p>
            {/* <p>{ nObj }</p> */}
            <p>{ nombre }</p>
            <p>{ JSON.stringify(nObj) }</p>
            <p>{ getSaludo('Arny') }</p>
            <p>{ null }</p>
            <p>{ numero * 2 }</p>
            <p>{ nombre }</p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    numero: PropTypes.number.isRequired
}


// SI define props aqui, se pueden usar en la app ej nombre
FirstApp.defaultProps = {
    // title: 'No hay titulo',
    numero: 0,
    nombre: 'Arny oculto'
}