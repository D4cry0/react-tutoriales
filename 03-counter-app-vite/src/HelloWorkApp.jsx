// Solo trabajar con functional components no clases

export function App() {
    // document.createElement....
    return (<h1>Hola Mundo!!</h1>);
}