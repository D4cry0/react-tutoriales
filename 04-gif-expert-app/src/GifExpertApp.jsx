import { useState } from "react";

import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

    // PUEDEN CONDICIONAR PORQUE PERDERIAN SU POSICIÓN Y REFERENCIA
    // REACT LOS ACOMODA EN ORDEN
    const [ categories, setCategories ] = useState([ 'Ghost In The Shell' ]);


    const onAddCategory = ( newCategory ) => {
        // push muta objetos y React no acepta las mutaciones por eso no se usa
        
        // Con desestructuración y operador spread se genera un nuevo arreglo
        // con eso se evito mutar el objeto anterios, entonces seria generar "copias" del objeto
        // inicializados con los datos anteriores y los nuevos
        
        if( categories.includes(newCategory) ) return;
        
        setCategories( [ newCategory, ...categories ] );
    }


    return (
        <>
            {/* titulo */}
            <h1>GfExpertApp</h1>

            {/* input */}
            <AddCategory 
                onNewCategory={ onAddCategory } 
            />

            {/* listado de cards */}
            {
                // Return implicito ()
                categories.map( category => (
                        <GifGrid key={ category } category={ category } />
                    ))
            }
            {/* gif item */}

        </>
    );
}
