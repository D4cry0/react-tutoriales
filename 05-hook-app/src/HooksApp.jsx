

export const HooksApp = () => {
    return (
        <>
            <div>Hooks</div>
        </>
    );
}
