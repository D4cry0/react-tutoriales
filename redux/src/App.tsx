import { useState, useEffect } from 'react'
import reactLogo from './assets/react.svg'
import viteLogo from '/vite.svg'
import './App.css'

import { useSelector, useDispatch } from 'react-redux'
import { RootState } from './store/store'
import { decrement, increment, incrementByAmount } from './store/slices/counter/counterSlice'
import { getPokemons } from './store/slices/pokemon/thunks'

function App() {

  const { value } = useSelector((state: RootState) => state.counter)
  const dispatch = useDispatch()
  

  return (
    <div className="App">
      <div>
        <a href="https://vitejs.dev" target="_blank">
          <img src={viteLogo} className="logo" alt="Vite logo" />
        </a>
        <a href="https://reactjs.org" target="_blank">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <h1>Vite + React</h1>
      <div className="card">
        <span>count is {value}</span>
        <button onClick={() => dispatch( increment() )}>
          +1
        </button>
        <button onClick={() => dispatch( incrementByAmount(4) )}>
          +4
        </button>
        <button onClick={() => dispatch( decrement() )}>
          -1
        </button>
        
      </div>

    </div>
  )
}

export default App
