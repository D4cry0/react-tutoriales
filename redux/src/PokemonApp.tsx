import React, { FC } from 'react'

export const PokemonApp: FC = () => {


    return (
        <>
            <h1>PokemonApp</h1>
            <hr />

            <ul>
                <li></li>
            </ul>
        </>
    )
}
