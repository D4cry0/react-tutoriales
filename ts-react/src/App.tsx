import React from 'react';

import { Counter } from './components/Counter';
import { Usuario } from './components/Usuario';

import { TimerPadre } from './components/TimerPadre';

import { ContadorRed } from './components/ContadorRed';
import { Formulario } from './components/Formulario';


import { Todo } from './todo/Todo';

function App() {
    return (
      <>
        <Todo />

      </>
    );
}

export default App;
