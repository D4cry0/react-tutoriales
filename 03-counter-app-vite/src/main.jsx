import React from 'react';
import ReactDOM from 'react-dom/client';

import { App } from './HelloWorkApp';
import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';

import './styles.css';

// title={ true } igual a title 

ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        {/* <FirstApp title="Desde el padre D:" numero={ 123 } /> */}
        <CounterApp value={ 10 } />
    </React.StrictMode>
);