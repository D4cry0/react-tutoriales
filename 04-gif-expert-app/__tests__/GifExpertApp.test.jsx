/**
 * @jest-environment jsdom
 */

import { fireEvent, render, screen } from "@testing-library/react";

import { GifExpertApp } from "../src/GifExpertApp";

describe('Pruebas en <GifExpertApp />', () => { 

    test('debe hacer match con el snapshot', () => { 
        const { container } = render( <GifExpertApp /> );
        
        expect( container ).toMatchSnapshot();
    });

    test('debe cambiar el valor del texto', () => { 
        render( <GifExpertApp /> );
        
        const input = screen.getByRole('textbox');
        fireEvent.input( input, { target: { value: 'Dreamcatcher' }});

        expect( input.value ).toBe('Dreamcatcher');
    });

    test('debe agregar el card item', () => { 
        render( <GifExpertApp /> );
        
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input( input, { target: { value: 'Dreamcatcher' }});
        fireEvent.submit( form );

        expect( screen.getByText('Dreamcatcher') ).toBeTruthy();
    });

    test('debe haber el mismo numero de card items a los terminos de busqueda', () => { 
        render( <GifExpertApp /> );
        
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input( input, { target: { value: 'Dreamcatcher' }});
        fireEvent.submit( form );
        fireEvent.input( input, { target: { value: 'BLack lagon' }});
        fireEvent.submit( form );

        // screen.debug();

        expect( screen.getAllByText('Cargando...').length ).toBe(3);
    });

});