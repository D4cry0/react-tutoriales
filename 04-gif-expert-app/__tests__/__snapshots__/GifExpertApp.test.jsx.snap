// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    GfExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifts"
      type="text"
      value=""
    />
  </form>
  <h3>
    Ghost In The Shell
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
